docs/sources/mix.md

```diff
@@ -0,0 +1,21 @@
+# Mix
+
+The mix source will detect `mix.lock` lockfiles, and parse their contents to enumerate dependencies.
+
+The `mix` CLI tool itself isn't used to extract dependency data, and no project compilation occurs during license extraction, but your dependencies should be present in your project `deps/` directory, as is conventional with Elixir projects.
+
+## Installing Dependencies
+
+Your project should contain a `deps/` directory containing the dependency sources. You can download your dependencies (which are defined in `mix.exs`) using `mix`:
+
+```
+mix deps.get
+```
+
+This will create your `mix.lock` lockfile if needed.
+
+Be sure to re-run this command whenever your `mix.exs` dependencies change to update your lockfile and dependency sources.
+
+## Limitations
+
+Because `mix deps.get` does not generate `mix.lock` entries for `:path` dependencies (nor are they stored in `deps/`), license information is not extracted from them.
```

Comment:
Hmmm are the `:path` dependencies enumerable from another source?  Not blocking to get this in, but if we're missing a class of dependencies entirely it's worth following up on

Reply:
In practice `:path` dependencies are used in development only, usually for debugging. (Not to say that people don't build locally from path deps and release, but I haven't personally seen it.)

While it is possible to get the list of deps from `mix.exs` (the project description, used to generate `mix.lock`) via something like:

```shell
$ mix run --no-start PATH/TO/SCRIPT"
```

Where the contents of the script would be something like the following (but with more error-handling logic):

```elixir
Mix.Dep.load_on_environment([]) # maybe specifically :prod
|> Enum.map(fn
  %{scm: Hex.SCM, status: {:ok, _}} = dep ->
    # Extract data
  %{scm: Mix.SCM.Path, status: {:ok, _}} = dep ->
    # Extract data
  %{scm: Mix.SCM.Git, status: {:ok, _}} = dep ->
    # Extract data
  dep ->
    # Generate error information
end)
|> format_for_parsing()
```

I'd suggest waiting until someone actively asks for `:path` support, as this introduces a lot of complexity/brittleness if Mix's internals change over time.

Reply:
One thing worth mentioning: if people use a local `:git` dependency vs `:path`, we get the definition in the `mix.lock`, e.g., changing the following in their `mix.exs`.

```
  {:absinthe, path: "/Users/bruce/code/absinthe-graphql/absinthe"}
``

to:

```
  {:absinthe, git: "/Users/bruce/code/absinthe-graphql/absinthe"}
```

Maybe I can suggest that as a possible work-around in the `mix.md` documentation?